pkg/azuredx/azureauth/auth_test.go

```diff
@@ -84,9 +85,13 @@ func TestQueryDataAuthorization(t *testing.T) {
 		}
 
 		// setup & test
-		c := &ServiceCredentials{HTTPDo: g.HTTPDoMock}
+		authority, err := resolveAuthorityForCloud(azsettings.AzurePublic)
+		require.NoError(t, err)
+
+		c := &ServiceCredentialsImpl{HTTPDo: g.HTTPDoMock}
 		c.TenantID = "0AF0528A-F473-4E0C-891F-3FF8ACDC4E5F"
 		c.OnBehalfOf = true
+		c.authority = authority
```

Comment:
```suggestion
		c := &ServiceCredentialsImpl{HTTPDo: g.HTTPDoMock}
		c.TenantID = "0AF0528A-F473-4E0C-891F-3FF8ACDC4E5F"
		c.OnBehalfOf = true
		c.authority = azidentity.AzurePublicCloud
```

Reply:
Right here I suggest "hardcoding" the `authority` you want in the the setup of your test rather than using a function to return it.